cmd/tikv-ctl/src/main.rs

```diff
@@ -2485,6 +2536,183 @@ fn run_ldb_command(cmd: &ArgMatches<'_>, cfg: &TiKvConfig) {
+        None => Vec::new(),
+    };
+    args.insert(0, "sst_dump".to_owned());
+    let opts = cfg.rocksdb.build_opt();
+    engine::rocks::run_sst_dump_tool(&args, &opts);
+}
+
+fn print_bad_ssts(db: &str, manifest: Option<&str>, pd_client: RpcClient, cfg: &TiKvConfig) {
+    let mut args = vec![
+        "sst_dump".to_string(),
+        "--output_hex".to_string(),
+        "--command=verify".to_string(),
+    ];
+    args.push(format!("--file={}", db));
+
+    let mut stderr = BufferRedirect::stderr().unwrap();
+    let stdout = BufferRedirect::stdout().unwrap();
+    let opts = cfg.rocksdb.build_opt();
+    match run_and_wait_child_process(|| engine::rocks::run_sst_dump_tool(&args, &opts)).unwrap() {
+        0 => {}
+        status => {
+            let mut err = String::new();
+            stderr.read_to_string(&mut err).unwrap();
+            println!("failed to run {}:\n{}", args.join(" "), err);
+            std::process::exit(status);
+        }
+    };
+
+    let mut stderr_buf = stderr.into_inner();
+    drop(stdout);
+    let mut buffer = Vec::new();
+    stderr_buf.read_to_end(&mut buffer).unwrap();
+    let mut corruptions = unsafe { String::from_utf8_unchecked(buffer) };
+
+    for line in corruptions.lines() {
+        println!("--------------------------------------------------------");
+        // The corruption format may like this:
+        // /path/to/db/057155.sst is corrupted: Corruption: block checksum mismatch: expected 3754995957, got 708533950  in /path/to/db/057155.sst offset 3126049 size 22724
+        println!("corruption info:\n{}", line);
+        let parts = line.splitn(2, ':').collect::<Vec<_>>();
+        let path = Path::new(parts[0]);
+        match path.extension() {
+            Some(ext) if ext.to_str().unwrap() == "sst" => {}
+            _ => {
+                println!("skip bad line format: {}", line);
+                continue;
+            }
+        }
+        let sst_file_number = path.file_stem().unwrap().to_str().unwrap();
+        let mut args1 = vec![
+            "ldb".to_string(),
+            "--hex".to_string(),
+            "manifest_dump".to_string(),
+        ];
+        args1.push(format!("--db={}", db));
+        args1.push(format!("--sst_file_number={}", sst_file_number));
+        if let Some(manifest_path) = manifest {
+            args1.push(format!("--manifest={}", manifest_path));
+        }
+
+        let mut stdout = BufferRedirect::stdout().unwrap();
+        let mut stderr = BufferRedirect::stderr().unwrap();
+        match run_and_wait_child_process(|| engine::rocks::run_ldb_tool(&args1, &opts)).unwrap() {
+            0 => {}
+            status => {
+                let mut err = String::new();
+                let mut stderr_buf = stderr.into_inner();
+                drop(stdout);
+                stderr_buf.read_to_string(&mut err).unwrap();
+                println!("failed to run {}:\n{}", args1.join(" "), err);
+                std::process::exit(status);
```

Comment:
We shouldn't directly exit here. Instead we should print out necessary information and continue.